analyzers/src/SonarAnalyzer.Common/Helpers/PropertiesHelper.cs

```diff
@@ -43,18 +44,38 @@ public static XElement[] ParseXmlSettings(SourceText sourceText)
         public static bool ReadIgnoreHeaderCommentsProperty(IEnumerable<XElement> settings, string language) =>
             ReadBooleanProperty(settings, language, "ignoreHeaderComments");
 
-        private static bool ReadBooleanProperty(IEnumerable<XElement> settings, string language, string propertySuffix, bool defaultValue = false)
-        {
-            var propertyLanguage = language == LanguageNames.CSharp ? "cs" : "vbnet";
-            var propertyName = $"sonar.{propertyLanguage}.{propertySuffix}";
-            return settings.Any()
-                && GetPropertyStringValue(propertyName) is { } propertyStringValue
+        public static string[] ReadSourceFileInclusionsProperty(IEnumerable<XElement> settings) =>
+            ReadCommaSeparatedArrayProperty(settings, "inclusions");
+
+        public static string[] ReadSourceFileExclusionsProperty(IEnumerable<XElement> settings) =>
+            ReadCommaSeparatedArrayProperty(settings, "exclusions");
+
+        private static bool ReadBooleanProperty(IEnumerable<XElement> settings, string language, string propertySuffix, bool defaultValue = false) =>
+            settings.Any()
+                && GetPropertyStringValue(settings, GetPropertyName(propertySuffix, language)) is { } propertyStringValue
                 && bool.TryParse(propertyStringValue, out var propertyValue)
                 ? propertyValue
                 : defaultValue;
 
-            string GetPropertyStringValue(string propName) =>
-                settings.FirstOrDefault(s => s.Element("Key")?.Value == propName)?.Element("Value").Value;
+        public static string[] ReadCommaSeparatedArrayProperty(IEnumerable<XElement> settings, string propertySuffix) =>
+            settings.Any()
+                && GetPropertyStringValue(settings, GetPropertyName(propertySuffix)) is { } propertyStringValue
+                && propertyStringValue.Split(',') is { } propertyValue
+                ? propertyValue
+                : new string[] { };
```

Comment:
There is no need to create a new object every time.
```suggestion
                : Array.Empty<string>();
```